app/feature/feature-claim-details/src/main/kotlin/com/hedvig/android/feature/claim/details/data/UploadFileUseCase.kt

```diff
@@ -0,0 +1,87 @@
+package com.hedvig.android.feature.claim.details.data
+
+import android.content.ContentResolver
+import android.net.Uri
+import arrow.core.Either
+import arrow.core.raise.either
+import arrow.core.raise.ensureNotNull
+import arrow.retrofit.adapter.either.networkhandling.CallError
+import com.hedvig.android.core.common.ErrorMessage
+import com.hedvig.android.core.common.android.FileService
+import com.hedvig.android.core.retrofit.toErrorMessage
+import com.hedvig.android.logger.LogPriority
+import com.hedvig.android.logger.logcat
+import kotlinx.serialization.Serializable
+import okhttp3.MediaType
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.MultipartBody
+import okhttp3.RequestBody
+import okio.BufferedSink
+import okio.source
+import retrofit2.http.Multipart
+import retrofit2.http.POST
+import retrofit2.http.Part
+import retrofit2.http.Query
+
+internal interface UploadFileUseCase {
+  suspend fun invoke(url: String, uri: Uri): Either<ErrorMessage, UploadSuccess>
+}
+
+internal data class UploadFileUseCaseImpl(
+  private val uploadFileService: UploadFileService,
+  private val fileService: FileService,
+  private val contentResolver: ContentResolver,
+) : UploadFileUseCase {
+  override suspend fun invoke(url: String, uri: Uri): Either<ErrorMessage, UploadSuccess> = either {
+    val claimId = Uri.parse(url).getQueryParameter("claimId") ?: raise(ErrorMessage("No claim id found in url"))
+    val result = uploadFileService.uploadFile(
+      claimId = claimId,
+      file = MultipartBody.Part.createFormData(
+        name = "files",
+        filename = fileService.getFileName(uri) ?: "media",
+        body = object : RequestBody() {
```

Comment:
I see that you use the same function here for sending documents, but also for sending photos that one may snap right at the moment.
In the chat code, I had this snippet https://github.com/HedvigInsurance/android/blob/ba69a697a6a38147da88510f53d08d2b107ce7c4/app/feature/feature-chat/src/main/kotlin/com/hedvig/android/feature/chat/data/ChatRepositoryImpl.kt#L124-L184 where `uploadFile` did things a tiny bit different for the photo scenario since we could just do a `uri.toFile()` to it and then upload it using the service + Retrofit.

But if this just works without it too, then this is a great opportunity to remove some extra code from inside ChatRepository too. Potentially share this part of the code between these two places, while still letting them upload to different services, as they need to. And even if those can not be shared, maybe the ChatRepository itself can reuse the file upload function for documents with the one which uploads the photo which was just taken.

But all this would then need to be extracted to a separate module which both of these can see, and would require some refactoring. Maybe not scoped for this PR, but it will long-term be a good idea not to have more or less the same code in two places.